compiler/noirc_frontend/src/elaborator/statements.rs

```diff
@@ -86,8 +86,8 @@ impl<'context> Elaborator<'context> {
                     expr_span,
                 }
             });
-            if annotated_type.is_unsigned() {
-                let errors = lints::overflowing_uint(self.interner, &expression, &annotated_type);
+            if annotated_type.is_unsigned() || annotated_type.is_signed() {
```

Comment:
Could this be an `is_integer` function? It appears that both `is_unsigned` and `is_signed` check for integers and then assert signed/unsigned.

```rust
matches!(self.follow_bindings(), Type::Integer(Signedness::Unsigned, _))
```

Reply:
Makes sense, done! I think I didn't do it because in another part of the code there's an integer type that's "field", but here it seems it's just those two.